proto/cometbft/types/v2/types.proto

```diff
@@ -0,0 +1,200 @@
+      [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  bytes validator_address = 6;
+  int32 validator_index   = 7;
+  // Vote signature by the validator if they participated in consensus for the
+  // associated block.
+  bytes signature = 8;
+  // Vote extension provided by the application. Only valid for precommit
+  // messages.
+  bytes extension = 9;
+  // Vote extension signature by the validator if they participated in
+  // consensus for the associated block.
+  // Only valid for precommit messages.
+  bytes extension_signature = 10;
+  // Non-Replay-Protected (NRP) vote extension provided by the application.
+  // Only valid for precommit messages.
+  bytes non_rp_extension = 11;
+  // Non-Replay-Protected (NRP) vote extension signature by the validator if
+  // they participated in consensus for the associated block.
+  // Only valid for precommit messages.
+  bytes non_rp_extension_signature = 12;
+}
+
+// Commit contains the evidence that a block was committed by a set of validators.
+message Commit {
+  int64              height     = 1;
+  int32              round      = 2;
+  BlockID            block_id   = 3 [(gogoproto.nullable) = false, (gogoproto.customname) = "BlockID"];
+  repeated CommitSig signatures = 4 [(gogoproto.nullable) = false];
+}
+
+// CommitSig is a part of the Vote included in a Commit.
+message CommitSig {
+  BlockIDFlag               block_id_flag     = 1;
+  bytes                     validator_address = 2;
+  google.protobuf.Timestamp timestamp         = 3
+      [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  bytes signature = 4;
+}
+
+// ExtendedCommit is a Commit with ExtendedCommitSig.
+message ExtendedCommit {
+  int64   height   = 1;
+  int32   round    = 2;
+  BlockID block_id = 3
+      [(gogoproto.nullable) = false, (gogoproto.customname) = "BlockID"];
+  repeated ExtendedCommitSig extended_signatures = 4 [(gogoproto.nullable) = false];
+}
+
+// ExtendedCommitSig retains all the same fields as CommitSig but adds vote
+// extension-related fields, where:
+// 'extension' and 'extension_signature' are used for replay-protected vote extensions.
+// 'non_rp_extension' and 'non_rp_extension_signature' are used for non-replay-protected vote extensions.
+// We use two signatures to ensure backwards compatibility.
+// That is the digest of the original signature is still the same in prior versions
+message ExtendedCommitSig {
+  BlockIDFlag               block_id_flag     = 1;
+  bytes                     validator_address = 2;
+  google.protobuf.Timestamp timestamp         = 3
+      [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  bytes signature = 4;
+  // Vote extension data
+  bytes extension = 5;
+  // Vote extension signature
+  bytes extension_signature = 6;
+  // Non-Replay-Protected vote extension data
+  bytes non_rp_extension = 7;
+  // Non-Replay-Protected vote extension signature
+  bytes non_rp_extension_signature = 8;
```

Comment:
Are we just adding a new field? If so, I think it's non-breaking, no?

Reply:
yes just adding fields in general. Non-breaking for go but for Rust where all fields need to be provided (see ABCI proto changes `VerifyVoteExtension`, `ExtendVote` RPCs).